### PR TITLE
Changes to support ROS 2

### PR DESCRIPTION
**Pull request by bponsler**
Initial port to get xacro working in ROS 2. Added function to generate a URDF file from a xacro file for use in ROS 2 launch files.

### CODE REVIEW

File: xacro/__init__.py [hunk 1]
```diff
@@ -1009,6 +1009,26 @@ def process_file(input_file_name, **kwargs):
 
     return doc
 
+def to_urdf(xacro_path, urdf_path=None):
```
Comment (rhaschke) [inline xacro/__init__.py:1012]: Why should we have this additional function? Shouldn't `ros_launch` call xacro as usual?

**Review by rhaschke — CHANGES_REQUESTED** [review]
What's the (general) plan to deal with ROS1 / ROS2 compatibility issues? This PR will break xacro on ROS1.

**Comment by bponsler** [comment]
Sorry for the delay in replying to your comments.

@rhaschke this PR was intended just to make these changes to support xacro in ros2 known to others and work to merge these changes into a single repo. The function was added because roslaunch works differently under ros2 (it is python) and I figured calling the python function directly would be better than using a system call to execute xacro but I’m not tied to a solution.

@codebot indeed that would be awesome. Not entirely sure how to handle the package.xml changes to go between ros1 and 2. Maybe it would be better as a pure python package instead of a ros package?

**Comment by traversaro** [comment]
> Maybe it would be better as a pure python package instead of a ros package? 

:+1: 

Transforming `xacro` in a pure python package would be also really convenient to install it on non-ROS friendly environment such as Windows, macOS or "strange" Linux distributions (see https://github.com/osrf/homebrew-simulation/issues/383).

**Comment by rhaschke** [comment]
I made ROS imports optional in #159. However, to make the package a pure python package, some more work needs to be done...

**Comment by gonzodepedro** [comment]
Working on port on #211

**Pull request closed without merging**